crypto/s2n_cipher.c

```diff
@@ -23,6 +26,11 @@ int s2n_session_key_alloc(struct s2n_session_key *key)
 {
     eq_check(key->evp_cipher_ctx, NULL);
     notnull_check(key->evp_cipher_ctx = EVP_CIPHER_CTX_new());
+#if defined(S2N_CIPHER_AEAD_API_AVAILABLE)
+    eq_check(key->evp_aead_ctx, NULL);
+    notnull_check(key->evp_aead_ctx = OPENSSL_malloc(sizeof(EVP_AEAD_CTX)));
```

Comment:
I thought about using `EVP_AEAD_CTX_new` but it expects the key so I see why you malloc this directly.

Reply:
Yes, and that the cipher suite is known.

Reply:
We usually try to prevent using memory allocation using OPENSSL_malloc, is this necessary here?

Reply:
In s2n, we try to prevent memory allocation using OPENSSL_malloc and use s2n's memory allocation,  is this necessary here?